ledger-core/virtual/integration/delegationtoken_test.go

```diff
@@ -297,3 +300,84 @@ func TestDelegationToken_IsMessageFromVirtualLegitimate(t *testing.T) {
 		})
 	}
 }
+
+func TestDelegationToken_CheckSuccessVDelegatedCallRequestFromOldPulseVEWithDT(t *testing.T) {
+	t.Log("C5186")
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	executeDone := server.Journal.WaitStopOf(&handlers.SMVDelegatedCallRequest{}, 1)
+
+	jetCoordinatorMock := jet.NewAffinityHelperMock(t)
+	auth := authentication.NewService(ctx, jetCoordinatorMock)
+	server.ReplaceAuthenticationService(auth)
+
+	var errorFound bool
+	logHandler := func(arg interface{}) {
+		if err, ok := arg.(error); ok {
+			errMsg := err.Error()
+			if strings.Contains(errMsg, "illegitimate msg") {
+				errorFound = true
+			}
+		}
+	}
+	logger := utils.InterceptLog(inslogger.FromContext(ctx), logHandler)
+	server.OverrideConveyorFactoryLogContext(inslogger.SetLogger(ctx, logger))
+
+	server.Init(ctx)
+
+	var (
+		class       = gen.UniqueGlobalRef()
+		outgoing    = server.BuildRandomOutgoingWithPulse()
+		executorRef = server.RandomGlobalWithPulse()
+		firstPulse  = server.GetPulse()
+	)
+
+	server.IncrementPulse(ctx)
+
+	approver := server.RandomGlobalWithPulse()
+	jetCoordinatorMock.
+		MeMock.Return(approver).
+		QueryRoleMock.Return([]reference.Global{approver}, nil)
+
+	delegationToken := server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), executorRef, outgoing)
```

Comment:
when i write case, i think in this case need send VDelegatedCallRequest twice and get delegation token from first response and replace in second request